src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -46,138 +46,104 @@ class CompanyStep extends React.Component {
+            errorTexts.incorporationDateFuture = this.props.translate('bankAccount.error.incorporationDateFuture');
+        }
 
-        return _.size(errors) === 0;
-    }
+        if (!values.incorporationState) {
+            errorTexts.incorporationState = this.props.translate('bankAccount.error.incorporationState');
+        }
 
-    submit() {
-        if (!this.validate()) {
-            return;
+        if (!values.companyPhone || !ValidationUtils.isValidUSPhone(values.companyPhone, true)) {
+            errorTexts.companyPhone = this.props.translate('bankAccount.error.phoneNumber');
         }
 
-        const bankAccount = {
-            // Fields from BankAccount step
-            ...ReimbursementAccountUtils.getBankAccountFields(this.props, ['bankAccountID', 'routingNumber', 'accountNumber', 'bankName', 'plaidAccountID', 'plaidAccessToken', 'isSavings']),
+        if (!values.companyName) {
+            errorTexts.companyName = this.props.translate('bankAccount.error.companyName');
+        }
 
-            // Fields from Company step
-            ...this.state,
-            incorporationDate: moment(this.state.incorporationDate).format(CONST.DATE.MOMENT_FORMAT_STRING),
-            companyTaxID: this.state.companyTaxID.replace(CONST.REGEX.NON_NUMERIC, ''),
-            companyPhone: LoginUtils.getPhoneNumberWithoutUSCountryCodeAndSpecialChars(this.state.companyPhone),
-        };
+        if (!values.hasNoConnectionToCannabis) {
+            errorTexts.hasNoConnectionToCannabis = this.props.translate('bankAccount.error.restrictedBusiness');
+        }
 
-        BankAccounts.updateCompanyInformationForBankAccount(bankAccount);
+        const errors = {};
+        _.each(errorTexts, (key) => {
+            errors[key] = true;
+        });
+
+        BankAccounts.setBankAccountFormValidationErrors(errors);
+        BankAccounts.updateReimbursementAccountDraft(values);
+
+        return errorTexts;
+    }
+
+    submit(values) {
+        const incorporationDate = moment(values.incorporationDate).format(CONST.DATE.MOMENT_FORMAT_STRING);
+        BankAccounts.setupWithdrawalAccount({
+            ...values,
+            incorporationDate,
+            companyTaxID: values.companyTaxID.replace(CONST.REGEX.NON_NUMERIC, ''),
+            companyPhone: LoginUtils.getPhoneNumberWithoutUSCountryCodeAndSpecialChars(values.companyPhone),
+        });
     }
 
     render() {
         const bankAccountID = ReimbursementAccountUtils.getDefaultStateForField(this.props, 'bankAccountID', 0);
         const shouldDisableCompanyName = bankAccountID && ReimbursementAccountUtils.getDefaultStateForField(this.props, 'companyName');
         const shouldDisableCompanyTaxID = bankAccountID && ReimbursementAccountUtils.getDefaultStateForField(this.props, 'companyTaxID');
 
+        FormActions.setErrors(ONYXKEYS.FORMS.COMPANY_STEP_FORM, this.props.reimbursementAccount.errors);
```

Comment:
Let's remove this. Form should automatically get the errors if we use the correct REIMBURSEMENT_ACCOUNT_FORM Onyx key
```suggestion

```

Reply:
done